x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/mistral/embeddings/MistralEmbeddingsServiceSettings.java

```diff
@@ -0,0 +1,208 @@
+
+        Integer dims = removeAsType(map, DIMENSIONS, Integer.class);
+        Boolean dimensionsSetByUser = extractOptionalBoolean(map, DIMENSIONS_SET_BY_USER, validationException);
+
+        switch (context) {
+            case REQUEST -> {
+                if (dimensionsSetByUser != null) {
+                    validationException.addValidationError(
+                        ServiceUtils.invalidSettingError(DIMENSIONS_SET_BY_USER, ModelConfigurations.SERVICE_SETTINGS)
+                    );
+                }
+                dimensionsSetByUser = dims != null;
+            }
+            case PERSISTENT -> {
+                if (dimensionsSetByUser == null) {
+                    validationException.addValidationError(
+                        ServiceUtils.missingSettingErrorMsg(DIMENSIONS_SET_BY_USER, ModelConfigurations.SERVICE_SETTINGS)
+                    );
+                }
+            }
+        }
+
+        if (validationException.validationErrors().isEmpty() == false) {
+            throw validationException;
+        }
+
+        return new MistralEmbeddingsServiceSettings(model, dims, dimensionsSetByUser, maxInputTokens, similarity, rateLimitSettings);
+    }
+
+    public MistralEmbeddingsServiceSettings(StreamInput in) throws IOException {
+        this.model = in.readString();
+        this.dimensions = in.readOptionalVInt();
+        this.dimensionsSetByUser = in.readBoolean();
+        this.similarity = in.readOptionalEnum(SimilarityMeasure.class);
+        this.maxInputTokens = in.readOptionalInt();
+        this.rateLimitSettings = new RateLimitSettings(in);
+    }
+
+    public MistralEmbeddingsServiceSettings(
+        String model,
+        @Nullable Integer dimensions,
+        Boolean dimensionsSetByUser,
+        @Nullable Integer maxInputTokens,
+        @Nullable SimilarityMeasure similarity,
+        @Nullable RateLimitSettings rateLimitSettings
+    ) {
+        this.model = model;
+        this.dimensions = dimensions;
+        this.dimensionsSetByUser = dimensionsSetByUser;
+        this.similarity = similarity;
+        this.maxInputTokens = maxInputTokens;
+        this.rateLimitSettings = Objects.requireNonNullElse(rateLimitSettings, DEFAULT_RATE_LIMIT_SETTINGS);
+    }
+
+    @Override
+    public String getWriteableName() {
+        return NAME;
+    }
+
+    @Override
+    public TransportVersion getMinimalSupportedVersion() {
+        return ADD_MISTRAL_EMBEDDINGS_INFERENCE;
+    }
+
+    public String model() {
+        return this.model;
+    }
+
+    public Integer dimensions() {
+        return this.dimensions;
+    }
+
+    public boolean dimensionsSetByUser() {
+        return this.dimensionsSetByUser;
+    }
+
+    public Integer maxInputTokens() {
+        return this.maxInputTokens;
+    }
+
+    public SimilarityMeasure similarity() {
+        return this.similarity;
+    }
+
```

Comment:
We also need to override the elementType method:


```
    @Override
    public DenseVectorFieldMapper.ElementType elementType() {
        return DenseVectorFieldMapper.ElementType.FLOAT;
    }
```

We really need to make it not default to null 🤦‍♂️ too easy to forget.